pkg/registry/extensions/deployment/strategy.go

```diff
@@ -77,6 +77,12 @@ func (deploymentStrategy) PrepareForUpdate(ctx api.Context, obj, old runtime.Obj
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Status = oldDeployment.Status
 
+	if newDeployment.Spec.Strategy.Type == extensions.RecreateDeploymentStrategyType {
```

Comment:
@lavalamp This is about making the API easier to use, such as with `kubectl apply`.

@ymqytw I'd make the test if Strategy.Type != RollingUpdate.

Do we allow unknown types? We've talked about that. See #31571. I'd like to find a way to both provide validation to prevent errors and to allow extensions.

cc @kargakis